hook/packet_monitor.c

```diff
@@ -0,0 +1,47 @@
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/netfilter.h>
+#include <linux/netfilter_ipv4.h>
+#include <linux/ip.h>
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Ouriel, Alexander");
+MODULE_DESCRIPTION("task 5");
+
+
+static unsigned int hook_func(void *priv, struct sk_buff *skb, const struct nf_hook_state *state) {
+    struct iphdr *ip_header;
+    // extract source and destination IP addresses from the packet
```

Comment:
Not accurate. IP header has more than src and dst.
Search for IP header and see what you can get from it.